src/PhpSpreadsheet/Chart/Renderer/JpGraphRendererBase.php

```diff
@@ -334,7 +334,9 @@ private function renderPlotLine(int $groupID, bool $filled = false, bool $combin
                 //    Set the appropriate plot marker
                 $this->formatPointMarker($seriesPlot, $marker);
             }
-            $dataLabel = $this->chart->getPlotArea()->getPlotGroupByIndex($groupID)->getPlotLabelByIndex($index)->getDataValue();
+            if ($this->chart->getPlotArea()->getPlotGroupByIndex($groupID)->getPlotLabelByIndex($index)) {
+                $dataLabel = $this->chart->getPlotArea()->getPlotGroupByIndex($groupID)->getPlotLabelByIndex($index)->getDataValue();
+            }
```

Comment:
I think `seriesPlot->SetLegend($dataLabel)` needs to be inside the if statement, otherwise dataLabel will not be defined.

Reply:
Also, please add a unit test to demonstrate that there was a problem before your fix and you've fixed it.